gui/packages/desktop/src/shared/ipc-event-channel.js

```diff
@@ -0,0 +1,155 @@
+// @flow
```

Comment:
FYI A lot of silly manual labour in here. I've tried to generate this code somehow but Flow can't really figure dynamically-built prototypes.